KIPs/kip-162.md

```diff
@@ -0,0 +1,204 @@
+
+### Txpool management
+
+A transaction pool should not accept underpriced transactions. A transaction is underpriced if its `tx.gasPrice` or `tx.maxFeePerGas` field is lower than the baseFeePerGas. The transaction pool can use last block's baseFeePerGas or the baseFeePerGas calculated for next block.
+
+## JSON-RPC API
+
+### `eth_gasPrice` and `klay_gasPrice`
+
+Returns suggested value for `gasPrice` or `maxFeePerGas` fields of a transaction.
+
+- Parameters: none
+- Result: Recommended gas price in peb. The result depends on the client implementation.
+- Example
+  ```sh
+  curl $RPC_URL -X POST -H 'Content-Type: application/json' --data '
+    {"jsonrpc":"2.0","id":1,"method":"eth_gasPrice","params":[]}'
+
+  {"jsonrpc":"2.0","id":1,"result":"0xba43b7400"}
+  ```
+
+### `eth_maxPriorityFeePerGas` and `klay_maxPriorityFeePerGas`
+
+Returns suggested value for `gasPrice` or `maxFeePerGas` fields of a transaction.
+
+- Parameters: none
+- Result: Recommended priority fee per gas in peb. The result depends on the client implementation.
+- Example
+  ```sh
+  curl $RPC_URL -X POST -H 'Content-Type: application/json' --data '
+    {"jsonrpc":"2.0","id":1,"method":"eth_maxPriorityFeePerGas","params":[]}'
+
+  {"jsonrpc":"2.0","id":1,"result":"0x3b9aca00"}
+  ```
+
+### `eth_feeHistory` and `klay_feeHistory`
+
+Returns historical gas information for a range of blocks.
+
+- Parameters
+  - `blockCount` - Number of blocks in the requested range.
+  - `newestBlock` - Highest block of the requested range. Can be a number or "latest".
+  - `rewardPercentiles` - (optional) A monotonically increasing list of percentile (between 0 and 100) values. For each block in the requested range, the transactions will be sorted in ascending order by effective tip per gas and sampled at the specified percentiles.
+- Result
+  - `oldestBlock` - Lowest number block of returned range.
+  - `baseFeePerGas` - An array of block base fee per gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-Magma blocks.
+  - `gasUsedRatio` - An array of block gas used ratios. Measures the network congestion level. These are calculated as the ratio of block gas used and [KIP-71 MAX_BLOCK_GAS_USED_FOR_BASE_FEE](https://github.com/klaytn/kips/blob/main/KIPs/kip-71.md). If the ratio is above 1, then 1 is returned.
+  - `reward` - A 2D array of effective priority fees per gas. `reward[n][i]` is the `rewardPercentiles[i]`-th percentile effective priority fees per gas among the transactions in the block `oldestBlock + n`. Only returned if `rewardPercentiles` is specified.
```

Comment:
```suggestion
  - `reward` - (optional) A 2D array of effective priority fees per gas. `reward[n][i]` is the `rewardPercentiles[i]`-th percentile effective priority fees per gas among the transactions in the block `oldestBlock + n`. Only returned if `rewardPercentiles` is specified.
```

Reply:
Fixed in https://github.com/klaytn/kips/commit/344af4e6cd4fa25a1104ad845a73d551a7acf7d3